Firestore/Source/API/FIRFirestore+Internal.h

```diff
@@ -68,6 +69,26 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (const std::shared_ptr<util::AsyncQueue> &)workerQueue;
 
+/**
+ * Attaches a listener for a snapshots-in-sync event. Server-generated
+ * updates and local changes can affect multiple snapshot listeners.
+ * The snapshots-in-sync event indicates that all listeners affected by
+ * a given change have fired.
+ *
+ * NOTE: The snapshots-in-sync event only indicates that listeners are
+ * in sync with each other, but does not relate to whether those
+ * snapshots are in sync with the server. Use SnapshotMetadata in the
+ * individual listeners to determine if a snapshot is from the cache or
+ * the server.
+ *
+ * @param listener A callback to be called every time all snapshot
+ * listeners are in sync with each other.
+ * @return A FIRListenerRegistration object that can be used to remove the
+ * listener.
+ */
+- (id<FIRListenerRegistration>)addSnapshotsInSyncListener:(void (^)())listener
```

Comment:
The type of the parameter should be `void (^)(void)`.

This declaration is intended to be part of the public Objective-C API, and that means that the type must conform to C's rules, not C++.

There are subtle differences between them: C treats the empty parameter list (i.e. `()`) as parameters unspecified. In C, to specify that the function actually takes no arguments you need to declare it as `foo(void)`. C++ has strict function prototypes and does not allow you to declare a function without declaring its parameters so in C++, `()` means no parameters.